ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/NoSameLineStatementAsOfBlockStartOrEnd.kt

```diff
@@ -0,0 +1,90 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.rule.engine.core.api.ElementType
+import com.pinterest.ktlint.rule.engine.core.api.IndentConfig
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.EditorConfig
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.INDENT_SIZE_PROPERTY
+import com.pinterest.ktlint.rule.engine.core.api.editorconfig.INDENT_STYLE_PROPERTY
+import com.pinterest.ktlint.rule.engine.core.api.indent
+import com.pinterest.ktlint.rule.engine.core.api.nextCodeLeaf
+import com.pinterest.ktlint.rule.engine.core.api.noNewLineInClosedRange
+import com.pinterest.ktlint.rule.engine.core.api.prevCodeLeaf
+import com.pinterest.ktlint.rule.engine.core.api.upsertWhitespaceAfterMe
+import com.pinterest.ktlint.rule.engine.core.api.upsertWhitespaceBeforeMe
+import com.pinterest.ktlint.ruleset.standard.StandardRule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+public class NoSameLineStatementAsOfBlockStartOrEnd
+    : StandardRule(
+    "no-same-line-statement-as-of-block-start-or-end",
+    usesEditorConfigProperties = setOf(
+        INDENT_SIZE_PROPERTY,
+        INDENT_STYLE_PROPERTY,
+    )
+) {
+    private var indentConfig = IndentConfig.DEFAULT_INDENT_CONFIG
+
+    override fun beforeFirstNode(editorConfig: EditorConfig) {
+        indentConfig =
+            IndentConfig(
+                indentStyle = editorConfig[INDENT_STYLE_PROPERTY],
+                tabWidth = editorConfig[INDENT_SIZE_PROPERTY],
+            )
+    }
+
+    override fun beforeVisitChildNodes(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        if (node.elementType == ElementType.FUN) {
+            node.findChildByType(ElementType.BLOCK)?.let { blockAstNode ->
+                val lBrace = blockAstNode.findChildByType(ElementType.LBRACE)!!
+                val rBrace = blockAstNode.findChildByType(ElementType.RBRACE)!!
+                val lBraceNextCodeElement = lBrace.nextCodeLeaf()!!
+                val rBracePrevCodeElement = rBrace.prevCodeLeaf()!!
+                val text = blockAstNode.text
+                val indexOfLBrace = text.indexOf("{")
+
+                assert(indexOfLBrace >= 0)
+
+                val indexOfRBrace = text.lastIndexOf("}")
+                assert(indexOfRBrace >= 0)
+
+                if (lBraceNextCodeElement.elementType == ElementType.RBRACE) {
+                    return
+                }
+
+                if (noNewLineInClosedRange(lBrace, lBraceNextCodeElement)) {
+                    emit(
+                        lBraceNextCodeElement.startOffset,
+                        "Expected new line after `{` of function body", true
+                    )
```

Comment:
Keep the emit on a single line when possible.

Reply:
Do we honour the max line?

Reply:
Yes mostly. Sometimes in test I suppress max line length for the lint violations.

Reply:
In this case, keeping emit in a single line was crossing the max len limit. Do we ignore max len for emit function to avoid clutter in PR?

Reply:
No, but then I would have expected the ktlintFormat task to put each parameter on a distinct line while in your code the second and third parameter are on the same line.

Reply:
Updated